### DISCUSSION TITLE
Add baby gender

### DISCUSSION THREAD

**Discussion by shalak** [babybuddy/babybuddy#885] [Ideas]
Can we consider adding gender to the child profile? It would help with translations, as most languages are gendered, and currently, they all default to the masculine grammatical gender.

**Comment by cdubz** [comment]
I am open to adding sex. But I'm unsure how to address this for translations generally. We use [POEditor](https://poeditor.com/join/project/QwQqrpTIzn) for the project and I don't see an obvious path to supporting that. Do you have any examples of how this is handled in other projects?

**Reply by shalak** [reply]
Unfortunately, I do not know how to enable this in POEditor. One way is to keep separate entries for all texts that refer to the baby (and in english they will be equal)

**Reply by mhogerheijde** [reply]
This might be doable by (mis)using ["message context"](https://www.gnu.org/software/gettext/manual/html_node/Contexts.html). Each translation could get a context with `msgctxt "masculine"`, `msgctxt "feminine"` and maybe `msgctxt "neutral"`. The call-site could then add the context to the request. This seems to be [supported](https://docs.djangoproject.com/en/5.2/topics/i18n/translation/#contextual-markers) by the django library in use.

POEditor also supports adding context to translations

![image](https://github.com/user-attachments/assets/4f23394f-7794-46ee-a9b5-ef44f1910a25)

Downsides I can quickly think of:
- I am not sure how this works when you request a term by its msgid, but no translation with the specific context is available. In other words, if a specific term has multiple contexts, do all translations need that translation with that context. In POEditor, this seems to be the case, you create a term with a context and all languages are requested to provide a translation for that term in that specific context. That means that all kind of terms need to be translated twice, even in languages where there is no gendered variant.
-  It seems that context is a single text-value, so if multiple context are needed, you'd need to come up with some way to combine context in a single string.
